### PR TITLE
Execute provided machine configuration and input

### PR DESCRIPTION
**Pull request by Devessier**

### CODE REVIEW

File: viz/src/machines/viz.ts [hunk 1]
```diff
@@ -1,79 +1,214 @@
...
                 },
               },
             },
-            "Automatic playing on": {
-              after: {
-                "automatic playing delay": {
-                  actions: "Increment step index",
-                  target: "Automatic playing on",
-                  internal: false,
+            "Managing machine and input execution": {
```
Comment (prastoin) [inline viz/src/machines/viz.ts:123]: praise: hehe

File: viz/src/machines/viz.ts [hunk 2]
```diff
@@ -1,79 +1,214 @@
...
+                    },
+                  },
+                  onDone: {
+                    target: "Fetched machine and input execution",
+                  },
+                },
+                "Fetched machine and input execution": {
+                  entry: "Allow to play execution steps",
+                  on: {
+                    Load: {
```
Comment (prastoin) [inline viz/src/machines/viz.ts:189]: question: it will start the machine execution inside the viz on a load hit ? what about a play or whatever action

Comment (prastoin) [reply viz/src/machines/viz.ts:189]: Ok nevermind, didn't see the entry action `Allow to play execution steps`

File: viz/src/machines/viz.ts [hunk 3]
```diff
@@ -1,79 +1,214 @@
...
+                              },
+                            ],
+                          },
+                        },
+                        "Received response": {
+                          type: "final",
+                        },
+                      },
+                    },
+                    "Delaying loading state": {
```
Comment (prastoin) [inline viz/src/machines/viz.ts:166]: I don't really understand this delay target here, unless I'm mistaken the submit button already manages his own delaying ?

File: viz/src/components/TheTape.vue [hunk 4]
```diff
@@ -38,59 +57,46 @@ const addedBlankSpace = squaresAmountOnOneSideOfHead;
...
  * there will always be the same amount of squares on the tape.
  */
-function fillStepsWithBlankSpacesOnSides(steps: TapeStep[]): TapeStep[] {
+function fillStepsWithBlankSpacesOnSides(steps: TapeStep[], blank: string): TapeStep[] {
   return steps.map(({ tape, ...props }) => ({
     tape: [
-      ...Array.from({ length: addedBlankSpace }).map(
-        () => blankCharacter.value
-      ),
+      ...Array.from({ length: addedBlankSpace }).map(() => blank),
```
Comment (prastoin) [inline viz/src/components/TheTape.vue:63]: Thoughts: I have no issue with the related code `array.from` + `map` are widely known and readable, just FYI `array.from` can take a fill function
```
const array = Array.from({ length: 5 }, () => blank);
```

File: viz/src/machines/viz.ts [hunk 5]
```diff
@@ -1,79 +1,214 @@
...
+      initial: "Application is ready",
       states: {
-        "Playing steps": {
-          initial: "Automatic playing off",
+        "Application is ready": {
+          invoke: {
+            src: "Start submit button machine",
+            id: "Submit button",
+          },
+          type: "parallel",
```
[the remark this answers is not in the document]
Comment (Devessier) [reply viz/src/machines/viz.ts:57]: Exactly!

File: viz/src/machines/viz.ts (hunk 3 above)
Comment (Devessier) [reply viz/src/machines/viz.ts:166]: The delay is made inside a child state of `Executing machine and input`. When the machine exits `Executing machine and input` state, the action `Exit loading state from submit button` is called, which sends an event to the machine that manages the load button to quit loading state. Thanks to this delay, the loading state is shown for at least one second. If the request takes more than one second, we do not wait more time and exit loading state right now.

Comment (prastoin) [reply viz/src/machines/viz.ts:166]: Ok I see !

**Pull request merged**